doc/architecture/decisions/0006-make-lists-a-thing.md

```diff
@@ -0,0 +1,106 @@
+
+### Asking questions for each thing in the list
+
+Early on in the development of runner the concept of repeating a group of questions based on answer count was implemented. This got us so far, but relies on isolating one answer to repeat on.
+Sometimes the list comes from more than one part of the questionnaire, for example there could be one set of repeats for collecting householders and another for collecting people who may be living away.
+
+## Decision
+
+- Define list creation in the schema
+- Store lists and answers within the context of a list
+- Remove all unnecessary repeat functionality
+
+### Define list creation in the schema
+
+A new block type will co-ordinate the collection of a list of things:
+
+```json
+{
+    "id": "block-id",
+    "type": "ListCollector",
+    "populates_list": "people",
+    "question": {},
+    "add_question": {},
+    "edit_question": {},
+    "remove_question": {}
+}
+```
+
+- The routing path is kept simple as we can ignore the journeys taken adding/editing/removing when evaluating the path.
+- Each question is a standard question type.
+- Each question can have variants. This is important as the add question can change depending on whether the list is empty or not.
+- More than one list collector can populate the same list.
+
+The user flows that will need to be implemented in runner from this schema:
+
+![Question Flow](0006-make-lists-a-thing-question-flow.png)
+
+### Store lists and answers within the context of a list
+
+A new top level item in the questionnaire store will store each named list and it's identifiers:
+
+```json
+{
+    "answers": [
+        {
+            "answer_id": "date-of-birth",
+            "value": "1982-08-01",
+            "repeat_identifier": "a9hd8j"
+        },
+        {
+            "answer_id": "occupation",
+            "value": "Teacher",
+            "repeat_identifier": "a9hd8j"
+        },
+        {
+            "answer_id": "date-of-birth",
+            "value": "1994-02-05",
+            "repeat_identifier": "kl9s3y"
+        },
+        {
+            "answer_id": "occupation",
+            "value": "Engineer",
+            "repeat_identifier": "kl9s3y"
+        }
+    ],
+    "lists": [
+        {
+            "name": "people",
+            "items": [
+                "a9hd8j",
+                "kl9s3y"
+            ]
+        }
+    ]
+}
+```
+
+- Removal of `group_instance`, `group_instance_id` and `answer_instance` from answers.
+- Introduction of a `repeat_identifier` that identifies the list item that an answer is related to.
+- The order of the items in the list definiton will determine display order.
+- The repeat identifier should be a short randomised string (it only has to be unique within the questionnaire).
```

Comment:
I worked this one out just for fun. `62^6` combinations so using `K^2 / 2N`, for 100 repeat identifiers per survey, the chance of collision = `8.8 * 10^-8`.

I guess we should still check it in any case